### DISCUSSION TITLE
Allow dry-run when testing new prune-settings

### DISCUSSION THREAD

**Discussion by schwma** [borgbase/vorta#855] [Ideas]
It would be great to be able to specify your own templates for archive names, just like when using borg backup from the CLI. For [example](https://borgbackup.readthedocs.io/en/stable/usage/create.html#examples):

`$ borg create /path/to/repo::{hostname}-{user}-{now} ~`.
`$ borg create /path/to/repo::{hostname}-{user}-{now:%Y-%m-%dT%H:%M:%S.%f} ~`

**Comment by m3nu** [comment]
Sure. We had long discussions on which archive name to use. Now that was all in vain... :-)

**Comment by ThomasWaldmann** [comment]
not in vain, it can be used as the default value still.

when making this adjustable, there needs to be also flexible prune prefix and maybe some sanity check, so people do not accidentally prune away lots of stuff.

**Comment by m3nu** [comment]
Here what I got so far. There is a preview and it won't save unless the formatting works.

Still thinking about the sanity check to avoid excessive pruning..

<img width="823" alt="screenshot 2018-12-11 at 23 15 18" src="https://user-images.githubusercontent.com/3916435/49809788-c499ba00-fd9a-11e8-850a-a4a20a793979.png">

**Comment by m3nu** [comment]
Left to do:

- [x] Actually use new settings for pruning and creating archives.
- [ ] Sanity check if possible
- [ ] Test

**Comment by ThomasWaldmann** [comment]
For borg cli, the sanity check is by the user reading the output of `borg prune --dry-run`.

**Comment by m3nu** [comment]
I ended up moving those 2 options to the Archives tab and putting the other pruning options in a separate section. Made more sense to keep them together. I'd also prefer to set new archive names per-profile instead of globally.

Will merge this as 0.6.1 (pre-release) soon, since there are 2 other bugs I'd like to see corrected. If you can think of any changes to improve this feature, let me know.

**Comment by m3nu** [comment]
<img width="833" alt="screenshot 2018-12-14 at 13 53 23" src="https://user-images.githubusercontent.com/3916435/49985744-077cad00-ffa8-11e8-92cb-fa9a9d525e07.png">

Screenshot of new *Prune Options* section. Left to do:

- [x] Actually use new settings for pruning and creating archives.
- [ ] Dry run and report on which repos would be pruned.
- [ ] Test
